### DISCUSSION TITLE
OpenAPI widget does not support nested objects

### DISCUSSION THREAD

**Discussion by kfc2023** [GitbookIO/community#132] [Feature Requests]
### What's your feature/idea?

When I use the OpenAPI widget to embed a method, the nested articles don't link to the associated models. This leaves out a huge component of my spec. Linking nested objects would be awesome. 

### What's your desired outcome?

Use gitbook as the sole source of documentation for my external api. right now, the details of the nested objects aren't accessible.

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by rachellawson** [comment]
This looks like something we are also needing.

**Comment by FiveOFive** [comment]
Sounds like what we need as well. Specifically the issue we're running into is that [references](https://swagger.io/docs/specification/using-ref/) and [inheritance](https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/) aren't included in the schema section of the OpenAPI widget. They do get included in the Example section.

**Reply by aikoii** [reply]
Hey @FiveOFive, AFAIK, the schema should be working with nested references and objects. If you could share (support@gitbook.com) the spec you are using, we can test it out 🙏 and see if there is a bug.

**Comment by kfc2023** [comment]
the value of the openapi widget is limited if you still have to provide a
link to the swagger output to provide all the info necessary to use the
method.

**Reply by aikoii** [reply]
Thanks for your input! We totally understand our OpenAPI block is not complete yet, and it's on our radar in the coming time🤞.
